backend/src/main/java/codezap/category/domain/Category.java

```diff
@@ -50,24 +55,24 @@ public class Category extends BaseTimeEntity {
     @Column(nullable = false)
     private Boolean isDefault;
 
-    public Category(String name, Member member) {
-        this.name = name;
-        this.member = member;
-        this.isDefault = false;
+    @Column(nullable = false)
+    private int ordinal;
+
+    public Category(String name, Member member, int ordinal) {
+        this(null, member, name, false, ordinal);
     }
 
     public static Category createDefaultCategory(Member member) {
-        return new Category(null, member, DEFAULT_CATEGORY_NAME, true);
+        return new Category(null, member, DEFAULT_CATEGORY_NAME, true, DEFAULT_CATEGORY_ORDINAL);
     }
 
-    public void updateName(String name) {
+    public void update(String name, int ordinal) {
         this.name = name;
+        this.ordinal = ordinal;
     }
```

Comment:
기본 카테고리의 이름이나 ordinal은 수정되면 안 된다고 생각해요.

드디어 여기서 도메인으로서의 검증을 할 수 있을 것 같네요. 

아래의 검증 로직이 update 실행 전에 호출되는 건 어떤가요?

메서드로 분리하는 것도 좋다고 생각해요. 

```suggestion
    public void update(String name, int ordinal) {
        if (isDefault) {
            throw new IllegalStateException("기본 카테고리는 수정할 수 없습니다.")
        }
        this.name = name;
        this.ordinal = ordinal;
    }
```

Reply:
카테고리 update를 변경감지가 아닌 update 쿼리를 직접 사용하도록 변경하였으므로 해당 코드는 삭제하였습니다!